### PR TITLE
Add -a, reading a puzzle from a list of candidates in the 927 format.

### PR DESCRIPTION
**Pull request by matjojo**
The test works on the basis of symmetry, we create Boards, get their candidates string, and create a board from that.

### CODE REVIEW

**Comment by matjojo** [comment]
After merging all tests still pass.

File: SudokuSolver/SolverFactory.cs [hunk 1]
```diff
@@ -28,6 +28,85 @@ public static Solver CreateBlank(int size, IEnumerable<string> constraints = nul
...
+                    {
+                        throw new ArgumentException($"ERROR: Could not parse a number in a candidates string: {maybeNumber}");
+                    }
+                    
+                }
+
+                sectionStartIndex += sectionLength;
+            }
+            while (sectionStartIndex < candidates.Length);
+
```
Comment (dclamage) [inline SudokuSolver/SolverFactory.cs:93]: Last night I added some additional functionality to remember which givens were initially provided. Please see the code that uses `isOriginalGiven` in `CreateFromGivens` and add it here as well.

File: SudokuSolverConsole/Program.cs [hunk 2]
```diff
@@ -150,6 +154,11 @@ static async Task Main(string[] args)
...
                 numBoardsSpecified++;
             }
+            if (haveCandidates)
+            {
+                numBoardsSpecified++;
+            }
+
             if (numBoardsSpecified != 1)
             {
                 Console.WriteLine($"ERROR: Cannot provide more than one set of givens (f-puzzles URL, given string, blank grid).");
```
Comment (dclamage) [inline SudokuSolverConsole/Program.cs:164]: Add "candidates" to the list in parenthesis.

File: SudokuSolverConsole/Program.cs [hunk 3]
```diff
@@ -177,10 +186,13 @@ static async Task Main(string[] args)
                 {
                     solver = SolverFactory.CreateFromGivens(givens, constraints);
                 }
-                else
+                else if (haveFPuzzlesURL)
                 {
                     solver = SolverFactory.CreateFromFPuzzles(fpuzzlesURL, constraints);
                     Console.WriteLine($"Imported \"{solver.Title ?? "Untitled"}\" by {solver.Author ?? "Unknown"}");
+                } else // if (haveCandidates)
```
Comment (dclamage) [inline SudokuSolverConsole/Program.cs:193]: [Style] Please make sure curly braces are always on their own line.

**Review by dclamage — CHANGES_REQUESTED** [review]
Thanks for the PR. Mostly looks good, just a couple small changes please.

The biggest one is that last night I did commit some modifications to `CreateFromGivens` which adds info the board about which givens were initially provided. I've added a review comment below for where that should appear.

Other than that, it's just minor style changes to match the rest of the codebase.

Thanks :)

File: SudokuSolver/SolverFactory.cs (hunk 1 above)
Comment (matjojo) [reply SudokuSolver/SolverFactory.cs:96]: Assuming that `isOriginalGiven` is only true for elements that have 1 and only 1 candidate I have this correctly implemented.

**Review by dclamage — APPROVED** [review]
Looks good, thanks!

**Pull request merged**